starport/pkg/cosmosgen/templates/vuex/store/index.ts.tpl

```diff
@@ -92,55 +103,59 @@ export default {
 		},
-		{{ range .Module.HTTPQueries }}async {{ .FullName }}({ commit, rootGetters, getters }, { options: { subscribe = false , all = false}, params: {...key}, query=null }) {
+		{{ range .Module.HTTPQueries }}
+		{{ $FullName := .FullName }}
+		{{ $Name := .Name }}
+		{{ range $i,$rule := .Rules}} 
+
+		async {{ $FullName }}{{ if (gt $i 0) }}{{$i}}{{ end}}({ commit, rootGetters, getters }, { options: { subscribe, all} = { subscribe:false, all:false}, params: {...key}, query=null }) {
 			try {
+				const queryClient=await initQueryClient(rootGetters)
+				let value= (await queryClient.{{ camelCase $FullName }}({{ range $j,$a :=$rule.Params}}{{ if (gt $j 0)}}, {{ end}} key.{{$a}}{{end}} {{ if $rule.HasQuery }}{{ if $rule.Params }}, {{ end }}query{{ end }}{{ if $rule.HasBody}}{{ if or $rule.HasQuery $rule.Params}},{{ end }}{...key}{{ end }} )).data
 				
-				let value = query?(await (await initQueryClient(rootGetters)).{{ camelCase .FullName }}({{ range $i,$a :=(index .Rules 0).Params}} key.{{$a}}, {{end}} query)).data:(await (await initQueryClient(rootGetters)).{{ camelCase .FullName }}({{ range $i,$a :=(index .Rules 0).Params}}{{ if (gt $i 0)}}, {{ end}} key.{{$a}} {{end}})).data
-				{{ if (index .Rules 0).HasQuery}}
+				{{ if $rule.HasQuery}}
 				while (all && (<any> value).pagination && (<any> value).pagination.nextKey!=null) {
-					let next_values=(await (await initQueryClient(rootGetters)).{{ camelCase .FullName }}({{ range $i,$a :=(index .Rules 0).Params}} key.{{$a}}, {{end}}{...query, 'pagination.key':(<any> value).pagination.nextKey})).data
-					for (let prop of Object.keys(next_values)) {
-						if (Array.isArray(next_values[prop])) {
-							value[prop]=[...value[prop], ...next_values[prop]]
-						}else{
-							value[prop]=next_values[prop]
-						}
-					}
+					let next_values=(await queryClient.{{ camelCase $FullName }}({{ range $j,$a :=$rule.Params}} key.{{$a}}, {{end}}{...query, 'pagination.key':(<any> value).pagination.nextKey}{{if $rule.HasBody}},{...key}{{ end}})).data
+					value = mergeResults(value, next_values);
 				}
 				{{ end }}
-				commit('QUERY', { query: '{{ .Name }}', key: { params: {...key}, query}, value })
-				if (subscribe) commit('SUBSCRIBE', { action: '{{ .FullName }}', payload: { options: { all }, params: {...key},query }})
-				return getters['get{{.Name }}']( { params: {...key}, query}) ?? {}
+				commit('QUERY', { query: '{{ $Name }}', key: { params: {...key}, query}, value })
+				if (subscribe) commit('SUBSCRIBE', { action: '{{ $FullName }}{{ if (gt $i 0) }}{{$i}}{{ end}}', payload: { options: { all }, params: {...key},query }})
+				return getters['get{{ $Name }}']( { params: {...key}, query}) ?? {}
 			} catch (e) {
-				console.error(new SpVuexError('QueryClient:{{ .FullName }}', 'API Node Unavailable. Could not perform query.'))
+				console.error(new SpVuexError('QueryClient:{{ $FullName }}{{ if (gt $i 0) }}{{$i}}{{ end}}', 'API Node Unavailable. Could not perform query: ' + e.message))
```

Comment:
index+1?

Reply:
I think leaving arbitrary logs in the generated code can be overwhelming since there is no way to disable it.

Error should be thrown instead to leave the option of catching the error to user and handle diffrently if desired.

Reply:
This was not a thrown error originally because it's more likely to be a runtime error for connectivity reasons rather than a coding error and a) I wanted to return something the application can work with b) not be forced to use await in a Vue component's mounting method for a snappier interface. Since dispatch is async, if it throws and it hasn't been awaited, we'll just get an "Uncaught error in promise()" log...

Still, I changed it to your suggestions (also adjusted subscription methods to catch as well)